src/plugins/explore/public/application/utils/state_management/slices/query/query_slice.ts

```diff
@@ -36,9 +36,19 @@ const querySlice = createSlice({
         meta: { addToHistory: true },
       }),
     },
+    setQueryStringWithHistory: {
+      reducer: (state, action: PayloadAction<string>) => {
+        // Same logic as setQueryState but with meta flag for history
+        state.query = action.payload;
```

Comment:
is it ok to modify state directly or does redux use immer or something

Reply:
yeah redux toolkit uses immer under the hood: https://redux-toolkit.js.org/usage/immer-reducers